scripts/find_packages.py

```diff
@@ -0,0 +1,121 @@
+        semi_pos = src.find(';')
+        if semi_pos > 0:
+            src = src[:semi_pos]
+        if src[0:7] == 'file://':
+            # TODO: Get full path of patches and other files within the source
+            # repo, not just the filesystem?
+            fetch = bb.fetch2.Fetch([], data)
+            local = fetch.localpath(src)
+            abcd_file.write('  - "%s"\n' % local)
+        else:
+            abcd_file.write('  - "%s"\n' % src)
+            if src[0:7] != 'http://' and src[0:8] != 'https://' and src[0:6] != 'ftp://' and src[0:6] != 'ssh://':
+                repos.append(src)
+    if len(repos) > 1:
+        print('Multiple repos not fully supported yet. Pacakge: %s' % info.pn)
+    for repo in repos:
+        colon_pos = repo.find(':')
+        abcd_file.write('  vcs:\n')
+        vcs_type = repo[:colon_pos]
+        if vcs_type == 'gitsm':
+            vcs_type = 'git'
+        abcd_file.write('    type: "%s"\n' % vcs_type)
+        abcd_file.write('    url: "%s"\n' % repo[colon_pos + 3:])
+        # TODO: Actually support multiple repos here:
+        abcd_file.write('    revision: "%s"\n' % srcrev)
+        abcd_file.write('    branch: "%s"\n' % branch)
+
+    abcd_file.write('  dependencies:\n')
+    for dep in depends:
+        abcd_file.write('  - "%s"\n' % dep)
+        # TODO: search for transitive dependencies here? Each dependency will
+        # get checked for its own dependencies sooner or later.
+
+    return depends
+
+
+def main():
+    abcd_manifest = 'manifest.abcd'
+    with open(abcd_manifest, "w") as abcd_file, bb.tinfoil.Tinfoil() as tinfoil:
+        tinfoil.prepare()
+        abcd_file.write('packages:\n')
+
+        # Does NOT include garage-sign, anything used only for testing (i.e.
+        # strace and gtest), any of the git submodules, all of which are also
+        # only used for testing (tuf-test-vectors, isotp-c, ostreesysroot,
+        # and HdrHistogram_c), or any other third party modules included
+        # directly into the source tree (jsoncpp, open62541, picojson)
+        recipes_to_check = ['aktualizr',
+                            'aktualizr-native',
+                            'aktualizr-auto-prov',
+                            'aktualizr-implicit-prov',
+                            'aktualizr-ca-implicit-prov',
+                            'aktualizr-hsm-prov',
+                            'aktualizr-disable-send-ip',
+                            'aktualizr-example-interface',
+                            'aktualizr-log-debug',
+                            'libp11', # BUILD_P11 (HSM) only
+                            'dpkg', # BUILD_DEB only
+                            'systemd'] # BUILD_SYSTEMD only
+
+        for recipe in recipes_to_check:
+            depends = print_deps(tinfoil, abcd_file, recipe)
+            for dep in depends:
+                if dep not in recipes_to_check:
+                    recipes_to_check.append(dep)
```

Comment:
Didn't expect that appending to a list inside while iterating on its item would work, but looks like it does. It feels a bit magical to me so perhaps it would warrant a small comment.

Reply:
Is it explicitly wrong/bad/undefined to do so? Should I use a queue instead?

Reply:
Well, the internet is full of people saying it's bad but actually the official docs mention this behavior: https://docs.python.org/3/tutorial/controlflow.html#for-statements. Didn't really find a "specification" though.

Reply:
Since I'm not deleting anything, I can't see why this would be a problem. If I use a queue, then I end up needing two data structures, a list to hold everything and a queue to hold just what still needs to be processed. Queues apparently also depend on threading, which I don't need. Deques and other structures don't really appear to be any better. For the moment, I'm hesitant to use something else if I don't have a good reason to.

Reply:
Yes, I think it's fine, the conciseness probably out-weights the cost here. I was just suggesting adding an explanatory comment.

I did write a version using a deque and a set but it actually took some tries to get it work correctly.

Reply:
Done (and rebased). I also tried to do it with a queue and realized it was quite ugly.